pyomo/util/slices.py

```diff
@@ -0,0 +1,289 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import six
+
+from pyomo.core.base.indexed_component import normalize_index
+from pyomo.core.base.indexed_component_slice import IndexedComponent_slice
+from pyomo.core.base.global_set import UnindexedComponent_set
+from pyomo.common.collections import ComponentSet, ComponentMap
+
+def _get_index_if_present(comp):
+    if hasattr(comp, 'index'):
+        return comp.index()
+    else:
+        return None
```

Comment:
I don't think this method is necessary.  See the comment in `get_component_call_stack`.

Reply:
You're right. Removed.